scripts/BorderControl.js

```diff
@@ -1,703 +1,764 @@
+    return;
+  }
+
+  static clamp(value, max, min) {
+    return Math.min(Math.max(value, min), max);
+  }
+  static newBorderColor() {
+    const overrides = {
+      CONTROLLED: {
+        INT: parseInt(game.settings.get('Border-Control', 'controlledColor').substr(1), 16),
+        EX: parseInt(game.settings.get('Border-Control', 'controlledColorEx').substr(1), 16),
+      },
+      FRIENDLY: {
+        INT: parseInt(game.settings.get('Border-Control', 'friendlyColor').substr(1), 16),
+        EX: parseInt(game.settings.get('Border-Control', 'friendlyColorEx').substr(1), 16),
+      },
+      NEUTRAL: {
+        INT: parseInt(game.settings.get('Border-Control', 'neutralColor').substr(1), 16),
+        EX: parseInt(game.settings.get('Border-Control', 'neutralColorEx').substr(1), 16),
+      },
+      HOSTILE: {
+        INT: parseInt(game.settings.get('Border-Control', 'hostileColor').substr(1), 16),
+        EX: parseInt(game.settings.get('Border-Control', 'hostileColorEx').substr(1), 16),
+      },
+      PARTY: {
+        INT: parseInt(game.settings.get('Border-Control', 'partyColor').substr(1), 16),
+        EX: parseInt(game.settings.get('Border-Control', 'partyColorEx').substr(1), 16),
+      },
+    };
+    if (this._controlled) return overrides.CONTROLLED;
+    else if (this._hover) {
+      let disPath = isNewerVersion(game.data.version, '0.8.0') ? CONST.TOKEN_DISPOSITIONS : TOKEN_DISPOSITIONS;
+      let d = parseInt(this.data.disposition);
+      if (!game.user.isGM && this.owner) return overrides.CONTROLLED;
+      else if (this.actor?.hasPlayerOwner) return overrides.PARTY;
+      else if (d === disPath.FRIENDLY) return overrides.FRIENDLY;
+      else if (d === disPath.NEUTRAL) return overrides.NEUTRAL;
+      else return overrides.HOSTILE;
+    } else return null;
+  }
+
+  static newTarget() {
+    const multiplier = game.settings.get('Border-Control', 'targetSize');
+    const INT = parseInt(game.settings.get('Border-Control', 'targetColor').substr(1), 16);
+    const EX = parseInt(game.settings.get('Border-Control', 'targetColorEx').substr(1), 16);
+
+    this.target.clear();
+    if (!this.targeted.size) return;
+
+    // Determine whether the current user has target and any other users
+    const [others, user] = Array.from(this.targeted).partition((u) => u === game.user);
+    const userTarget = user.length;
+
+    // For the current user, draw the target arrows
+    if (userTarget) {
+      if (game.settings.get('Border-Control', 'internatTarget')) {
+        let p = -4; // padding
+        let aw = -12 * multiplier; // arrow width
+        let h = this.h; // token height
+        let hh = h / 2; // half height
+        let w = this.w; // token width
+        let hw = w / 2; // half width
+        let ah = (canvas.dimensions.size / 3) * multiplier;
+        this.target
+          .beginFill(INT, 1.0)
+          .lineStyle(1, EX)
+          .drawPolygon([-p - aw, hh, -p, hh - ah, -p, hh + ah])
```

Comment:
I explicitly setout the polygon arrays to make then more human-readable rather than "neater"